uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/components/CheckBoxItem.kt

```diff
@@ -26,48 +26,53 @@ import tech.dojo.pay.uisdk.presentation.components.theme.DojoTheme
 internal fun CheckBoxItem(
     modifier: Modifier = Modifier,
     itemText: String,
-    onCheckedChange: (Boolean) -> (Unit),
+    isChecked: Boolean = true,
+    onCheckedChange: (Boolean) -> Unit,
 ) {
-    val checkedState = remember { mutableStateOf(true) }
-    Box(modifier = modifier) {
-        Row(
-            modifier = Modifier
-                .fillMaxWidth()
-                .height(30.dp),
-            verticalAlignment = Alignment.CenterVertically
+    val checkedState = remember { mutableStateOf(isChecked) }
 
+    Row(
+        modifier = modifier
+            .fillMaxWidth()
+            .clickable {
+                checkedState.value = !checkedState.value
+                onCheckedChange(checkedState.value)
+            },
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        Box(
+            modifier = Modifier
+                .size(25.dp)
+                .border(
+                    width = 1.dp,
+                    color = if (checkedState.value) DojoTheme.colors.inputElementActiveTintColor else DojoTheme.colors.inputElementDefaultTintColor,
+                    shape = DojoTheme.shapes.small,
+                )
+                .background(DojoTheme.colors.primarySurfaceBackgroundColor)
+                .align(Alignment.Top),
```

Comment:
`TriStateCheckbox` is not giving the same UI as what i need

Reply:
We should always aim for using the components android give us instead of building new ones for small layout changes. Make sure next time you try to push for using the Android components rather than creating new one